rpcs3/Emu/Cell/Modules/cellSsl.cpp

```diff
@@ -66,12 +78,15 @@ s32 cellSslCertificateLoader(u64 flag, vm::ptr<char> buffer, u32 size, vm::ptr<u
 		{
 			if (!flagBits[i-1])
 				continue;
-			final.append(getCert(certPath, i));
+			// If we're loading cert 6 (the baltimore cert), then we need set that we're loading the 'normal' set of certs.
+			if (flagBits[5])
+				final.append(getCert(certPath, i, true));
+			else
+				final.append(getCert(certPath, i));
 		}
 
-		memset(buffer.get_ptr(), 0, size);
-		strncpy(buffer.get_ptr(), final.c_str(), size);
-		buffer.get_ptr()[size] = '\0';
+		memset(buffer.get_ptr(), '\0', size - 1);
+		memcpy(buffer.get_ptr(), final.c_str(), final.size());
```

Comment:
Btw, is `final.size()` always less than `size`?

Reply:
yes